com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/DebugAdapter.java

```diff
@@ -113,6 +116,19 @@ private void initialize() {
     }
 
     private void registerHandler(IDebugRequestHandler handler) {
+        registerHandler(handler, true, false);
+    }
+
+    private void registerHandler(IDebugRequestHandler handler, boolean forDebug, boolean forNoDebug) {
```

Comment:
Two boolean value is tedious for the API level.  Either one boolean value or enum value.  

Keep the function/method declaration succinct.